cpp/include/Ice/PluginFactory.h

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) ZeroC, Inc.
+
+#ifndef ICE_PLUGIN_FACTORY_H
+#define ICE_PLUGIN_FACTORY_H
+
+#include "CommunicatorF.h"
+#include "Config.h"
+#include "Ice/BuiltinSequences.h"
+#include "Plugin.h"
+
+#include <string>
+
+namespace Ice
+{
+    class Plugin;
+
+    /// A plug-in factory function is a C function responsible for creating an Ice plug-in.
```

Comment:
I'm not sure about this "C function" part. I guess this is only relevant for dynamic loaded plug-ins.

Reply:
All the plug-in create function are extern "C" in C++, see:
https://doc.zeroc.com/ice/3.7/communicator-and-other-core-local-features/plug-in-facility/plug-in-api#id-.PluginAPIv3.7-C++Plug-inFactory